trace/kafka/option_test.go

```diff
@@ -90,3 +94,32 @@ func TestRequiredAcksPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestEncoder(t *testing.T) {
+	type args struct {
+		enc encoding.EncodeFunc
+		ct  string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{name: "json EncodeFunc", args: args{enc: json.Encode, ct: json.Type}, wantErr: false},
+		{name: "protobuf EncodeFunc", args: args{enc: protobuf.Encode, ct: protobuf.Type}, wantErr: false},
+		{name: "nil EncodeFunc", args: args{enc: nil}, wantErr: true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := sarama.NewConfig()
+			ap := &AsyncProducer{cfg: cfg}
+			err := Encoder(tt.args.enc, tt.args.ct)(ap)
+			if tt.wantErr {
```

Comment:
it would ge good to check that these values are actually set in the producer

Reply:
Done in the `else` clause; sorry for not replying earlier.